### PR TITLE
[VYMCA-117] Improve favorites performance

### PR DESCRIPTION
**Pull request by hamrant**
**Related Issue/Ticket:**

In this PR was fixed some of comments from @podarok in https://github.com/ymcatwincities/openy_gated_content/pull/88

For favorites performance we have:
- Caching per user + invalidation on add/remove item
- add/remove item endpoints not return list (items handled on vuex side)

## Steps to test:

- [ ] Check that favorites works as before this fix

### CODE REVIEW

File: modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/YUSAClientService.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ class YUSAClientService {
    *
    * @var \Drupal\Core\Logger\LoggerChannelFactory
```
Comment (NightHunterSV) [inline modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/YUSAClientService.php:19]: It will be great to correct the class or interface here

**Review by NightHunterSV — APPROVED** [review]

File: src/Controller/FavoritesController.php [hunk 2]
```diff
@@ -143,23 +173,21 @@ public function delete(Request $request) {
     if ($data instanceof JsonResponse) {
       return $data;
     }
+    $data['uid'] = $this->currentUser()->id();
     $favorite_items = $this->entityTypeManager()
       ->getStorage('vy_favorite_item')
-      ->loadByProperties([
-        'uid' => $this->currentUser()->id(),
```
Comment (anpolimus) [inline src/Controller/FavoritesController.php:149]: Does it mean that I could update other user data from frontend if I know his uid?

**Review by anpolimus — CHANGES_REQUESTED** [review]

File: src/Controller/FavoritesController.php [hunk 3]
```diff
@@ -115,14 +145,14 @@ public function add(Request $request) {
 
     $favorite_item = $this->entityTypeManager()
       ->getStorage('vy_favorite_item')
-      ->create([
-        'ref_entity_type' => $data['entity_type'],
-        'ref_entity_bundle' => $data['entity_bundle'],
-        'ref_entity_id' => $data['entity_id'],
-      ]);
+      ->create($data);
```
Comment (NightHunterSV) [inline src/Controller/FavoritesController.php:148]:
```
fetch("http://carnation.pr-231.openy-gated-content.sandbox-2.ci.fivejars.com/api/favorites/add?_format=json", {
  "headers": {
    "accept": "application/json, text/plain, */*",
    "accept-language": "ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7,uk;q=0.6",
    "content-type": "application/json"
  },
  "referrer": "http://carnation.pr-231.openy-gated-content.sandbox-2.ci.fivejars.com/virtual-ymca",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": "{\"uid\":16,\"ref_entity_type\":\"node\",\"ref_entity_bundle\":\"gc_video\",\"ref_entity_id\":3}",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});
```
Allowed me to "like" the item for another user. Let's return to the previous implementation, or, at least, unset the `uid` from an input data.

**Review by NightHunterSV — CHANGES_REQUESTED** [review]

File: src/Controller/FavoritesController.php (hunk 2 above)
Comment (hamrant) [reply src/Controller/FavoritesController.php:149]: on 176 line we have `$data['uid'] = $this->currentUser()->id();`

File: src/Controller/FavoritesController.php (hunk 3 above)
Comment (hamrant) [reply src/Controller/FavoritesController.php:148]: Fixed in https://github.com/fivejars/openy_gated_content/pull/231/commits/9289916fa04cd0719ecd5870623170c62ed8cf87

File: modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/YUSAClientService.php (hunk 1 above)
Comment (hamrant) [reply modules/openy_gc_auth/modules/openy_gc_auth_yusa/src/YUSAClientService.php:19]: It will be great, but not today! :stuck_out_tongue:

**Review by NightHunterSV — APPROVED** [review]

**Comment by hamrant** [comment]
retest

**Review by anpolimus — APPROVED** [review]

**Comment by NadyaRudenko** [comment]
tested, all good. Thank you!

**Pull request merged**